src/tribler/core/utilities/tiny_tribler_service.py

```diff
@@ -72,13 +75,16 @@ async def _start_session(self):
 
     def _check_already_running(self):
         self.logger.info(f'Check if we are already running a Tribler instance in: {self.config.state_dir}')
-
         root_state_dir = get_root_state_directory(create=True)
-        current_process = TriblerProcess.current_process(ProcessKind.Core)
-        self.process_manager = ProcessManager(root_state_dir, current_process)
+
+        self.process_lock = try_acquire_file_lock(root_state_dir / CORE_LOCK_FILENAME)
+        current_process_has_lock = bool(self.process_lock)
+
+        self.process_manager = ProcessManager(root_state_dir)
+        self.process_manager.setup_current_process(kind=ProcessKind.Core, has_lock=current_process_has_lock)
```

Comment:
I've encountered these lines for the third time now. Is there any chance they could be extracted or generalized?

Reply:
Agreed. I added the `setup_process_manager` utility function that manages this initialization logic.